templates/partials/contacts_content.html

```diff
@@ -125,7 +125,7 @@
                 </div>
               </div>
               <div ng-show="selected.doc.type === 'clinic'" mm-auth="can_view_tasks" class="col-xs-1 tasks-count">
-                <span class="badge">{{ child.doc.tasks.length || 0 }}</span>
+                <span class="badge">{{ (selected.tasks | filter:filterTasksByChild(child)).length || 0 }}</span>
```

Comment:
Also, you can include a filter object directly instead of using a function. This has pros and cons (including logic in templates) but I think on balance I slightly prefer it. This will need testing but I think something like this would just work:

```js
{{ (selected.tasks | filter:{ doc: { contact: { _id: child.doc._id }}}).length }}
```